src/textual/command_palette.py

```diff
@@ -0,0 +1,864 @@
+        flag's state.
+        """
+        self.query_one(LoadingIndicator).set_class(self._show_busy, "--visible")
+        self.query_one(CommandList).set_class(self._show_busy, "--populating")
+
+    @staticmethod
+    async def _consume(
+        source: CommandMatches, commands: Queue[CommandSourceHit]
+    ) -> None:
+        """Consume a source of matching commands, feeding the given command queue.
+
+        Args:
+            source: The source to consume.
+            commands: The command queue to feed.
+        """
+        async for hit in source:
+            await commands.put(hit)
+
+    async def _search_for(
+        self, search_value: str
+    ) -> AsyncGenerator[CommandSourceHit, bool]:
+        """Search for a given search value amongst all of the command sources.
+
+        Args:
+            search_value: The value to search for.
+
+        Yields:
+            The hits made amongst the registered command sources.
+        """
+
+        # Get the style for highlighted parts of a hit match.
+        match_style = self._sans_background(
+            self.get_component_rich_style("command-palette--highlight")
+        )
+
+        # Set up a queue to stream in the command hits from all the sources.
+        commands: Queue[CommandSourceHit] = Queue()
+
+        # Fire up an instance of each command source, inside a task, and
+        # have them go start looking for matches.
+        assert self._calling_screen is not None
+        searches = [
+            create_task(
+                self._consume(
+                    source(self._calling_screen, match_style).search(search_value),
+                    commands,
+                )
+            )
+            for source in self._sources
+        ]
+
+        # Set up a delay for showing that we're busy.
+        self._start_busy_countdown()
+
+        # Assume the search isn't aborted.
+        aborted = False
+
+        # Now, while there's some task running...
+        while not aborted and any(not search.done() for search in searches):
+            try:
+                # ...briefly wait for something on the stack. If we get
+                # something yield it up to our caller.
+                aborted = yield await wait_for(commands.get(), 0.1)
+            except TimeoutError:
+                # A timeout is fine. We're just going to go back round again
+                # and see if anything else has turned up.
+                pass
+            except CancelledError:
+                # A cancelled error means things are being aborted.
+                aborted = True
+            else:
+                # There was no timeout, which means that we managed to yield
+                # up that command; we're done with it so let the queue know.
+                commands.task_done()
+
+        # Check through all of the finished searches and, if any of them has
+        # an exception, bubble it up.
+        search_exception = next(
```

Comment:
Normally I'd like the approach of bailing out when an error occurs.

For this use case though, I think it may be better to log the exceptions rather than have the app exit.

My reasoning is that we don't want a single broken command provider to make the command palette unusable.

Reply:
Done.